### PR TITLE
ci: Correct some drift between macOS and ubuntu builds

### LINKED ISSUE #63: Investigate CI failure on MacOS 10.15
I can build on my Mojave machine (10.14) but CI dies on Catalina (10.15) with:
```
ld: framework not found -lCocoa
```

### PR DESCRIPTION
**Pull request by ddn0**
Closes #63

### CODE REVIEW

**Comment by witchel** [comment]
This is probably not the right spot, but I am wondering about the Catalina build environment as I will be switching to laptop use for a few days.  I'm not sure if Issue #63 means we have MacOS build issues.  I don't see a setup_dev_macos.sh, in analogy with setup_dev_ubuntu.sh.

**Comment by ddn0** [comment]
> This is probably not the right spot, but I am wondering about the Catalina build environment as I will be switching to laptop use for a few days. I'm not sure if Issue #63 means we have MacOS build issues. I don't see a setup_dev_macos.sh, in analogy with setup_dev_ubuntu.sh.

You read my mind!

I'm thinking of reorganizing this PR to re-enable the MacOS CI and bring all the setup inline with our Linux builds. I was able to build on a Catalina machine so I think #63 must have to do with how the GitHub runner is configured rather than Catalina itself.

**Comment by ddn0** [comment]
@scober `mmap(..., MAP_32BIT)` is not available on MacOS. I just removed it, but let me know if I should attempt a better workaround.

**Comment by scober** [comment]
> Open questions:
> 
> - @scober `mmap(..., MAP_32BIT)` is not available on MacOS. I just removed it, but let me know if I should attempt a better workaround.
> 

@patrickkenney9801  was going to remove this too as it was causing OOM errors in large graph conversions. It is not necessary for correctness, just an ill fated performance hack.

**Comment by patrickkenney9801** [comment]
yes, removing MAP_32BIT is also in my current PR

**Comment by tylershunt** [comment]
> * I'm tempted to enable the MacOS builds for now but disable the tests until we can reproduce this more easily.

That seems reasonable to me.

File: libtsuba/tests/md5.cpp [hunk 1]
```diff
@@ -7,7 +7,9 @@
 
 #include "md5.h"
 
-#ifndef _MSC_VER
+#if defined(__APPLE__)
+#include <machine/endian.h>
+#elif !defined(_MSC_VER)
```
Comment (tylershunt) [inline libtsuba/tests/md5.cpp:12]: Just curious: Does the build work with `cl`? I've never tried it.

Comment (ddn0) [reply libtsuba/tests/md5.cpp:12]: @insertinterestingnamehere has probably tried most recently. I believe the current status is the build doesn't work on Windows.

**Comment by arthurp** [comment]
I can follow this PR with a PR to enable MacOS conda packaging. Once builds on MacOS are working Conda packaging should be straight forward.

**Pull request merged**